Sources/AWSSDKSwiftCore/HTTP/HTTPClient.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  HTTPClient.swift
+//  AWSSDKSwiftCore
+//
+//  Created by Adam Fowler on 2019/11/8
+//
+
+import Foundation
+import NIO
+import NIOHTTP1
+
+/// HTTP Request
+struct AWSHTTPRequest {
+    let url: URL
+    let method: HTTPMethod
+    let headers: HTTPHeaders
+    let bodyData: Data?
+}
+
+/// HTTP Response
+protocol AWSHTTPResponse {
+    var status: HTTPResponseStatus { get }
+    var headers: HTTPHeaders { get }
+    var bodyData: Data? { get }
```

Comment:
Just out of curiosity: Why do we use `Foundation.Data` and not `NIO.ByteBuffer`?

Reply:
We were already generating `Data` objects for requests. If we were to move to`NIO.ByteBuffer` I would want to make sure we aren't making additional copies of the body data when moving between data types. I just haven't had time to look into that.

Reply:
Okay. No problem at all!

Reply:
done now